packages/test-schemas/src/media.ts

```diff
@@ -1,36 +1,120 @@
+interface ImageMedia implements MediaMetadata & VisualMedia @createModel(description: "An interface for image objects") {
   src: String! @string(maxLength: 500)
   size: Int
   width: Int! @int(min: 1)
   height: Int! @int(min: 1)
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
 }
 
-type AudioMedia implements MediaMetadata & TimeMedia @createModel(accountRelation: LIST, description: "A model for audio objects") {
+interface AudioMedia implements MediaMetadata & TimeMedia @createModel(description: "An interface for audio objects") {
   src: String! @string(maxLength: 500)
   size: Int
-  duration: Int! @int(min: 0)
+  duration: Duration!
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
 }
 
-type VideoMedia implements MediaMetadata & VisualMedia & TimeMedia @createModel(accountRelation: LIST, description: "A model for image objects") {
+interface VideoMedia implements MediaMetadata & VisualMedia & TimeMedia @createModel(description: "An interface for video objects") {
   src: String! @string(maxLength: 500)
   size: Int
-  duration: Int! @int(min: 0)
+  duration: Duration!
   width: Int! @int(min: 1)
   height: Int! @int(min: 1)
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
+}
+
+type MyImage implements ImageMedia
+  @createModel(description: "A model for image objects")
+  @createIndex(fields: [{ path: ["src"] }])
+  @createIndex(fields: [{ path: ["width"] }])
+  @createIndex(fields: [{ path: ["height"] }]) {
+  src: String! @string(maxLength: 500)
+  size: Int
+  width: Int! @int(min: 1)
+  height: Int! @int(min: 1)
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
+}
+
+type MyAudio implements AudioMedia
+  @createModel(description: "A model for audio objects")
+  @createIndex(fields: [{ path: ["duration"] }]) {
+  src: String! @string(maxLength: 500)
+  size: Int
+  duration: Duration!
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
+}
+
+type MyVideo implements VideoMedia
+  @createModel(description: "A model for video objects")
+  @createIndex(fields: [{ path: ["src"] }])
+  @createIndex(fields: [{ path: ["duration"] }])
+  @createIndex(fields: [{ path: ["width"] }])
+  @createIndex(fields: [{ path: ["height"] }]) {
+  src: String! @string(maxLength: 500)
```

Comment:
do we have a test for interfaces with conflicting field definitions?  should that throw an error?

Reply:
No test yet as I've only focused on the "happy paths" so far but I'll add coverage for these kinds of checks